### PR TITLE
feat:add consumer override of quotas

### PR DESCRIPTION
**Pull request by ericyz**
There are quite a few use cases to add customer quota override during the project creation process. The changes is to add the capability into the CFT module

### CODE REVIEW

**Review by morgante — CHANGES_REQUESTED** [review]

**Comment by ocervell** [comment]
I'd love to see this land. @ericyz would you please address @morgante 's comments so that we can add this to Project Factory ?

**Comment by ericyz** [comment]
Oh, I missed out this one. For sure, I am updating it now

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/README.md /tmp/tmp.IzTRERkDLd/generate_docs/workspace/README.md
123a124
> | consumer\_quotas | The quotas configuration you want to override to the project. | object | `<list>` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate . 
Success!
 The configuration is valid.
terraform_validate ./examples/app_engine 
Success!
 The configuration is valid.
terraform_validate ./examples/budget_project 
Success!
 The configuration is valid.
terraform_validate ./examples/fabric_project 
Success!
 The configuration is valid.
terraform_validate ./examples/gke_shared_vpc 
Success!
 The configuration is valid.
terraform_validate ./examples/group_project 
Provider "gsuite" not available for installation.
A provider named "gsuite" could not be found in the Terraform Registry.
This may result from mistyping the provider name, or the given provider may
be a third-party provider that cannot be installed automatically.
In the latter case, the plugin must be installed manually by locating and
downloading a suitable distribution package and placing the plugin's executable
file in the following directory:
    terraform.d/plugins/linux_amd64
Terraform detects necessary plugins by inspecting the configuration and state.
To view the provider versions requested by each module, run
"terraform providers".
Error: 
no provider exists with the given name
terraform_validate ./examples/project-hierarchy 
Provider "gsuite" not available for installation.
A provider named "gsuite" could not be found in the Terraform Registry.
This may result from mistyping the provider name, or the given provider may
be a third-party provider that cannot be installed automatically.
In the latter case, the plugin must be installed manually by locating and
downloading a suitable distribution package and placing the plugin's executable
file in the following directory:
    terraform.d/plugins/linux_amd64
Terraform detects necessary plugins by inspecting the configuration and state.
To view the provider versions requested by each module, run
"terraform providers".
Error: 
no provider exists with the given name
terraform_validate ./examples/project_services 
Success!
 The configuration is valid.
terraform_validate ./examples/shared_vpc 
Success!
 The configuration is valid.
terraform_validate ./examples/simple_project 
Success!
 The configuration is valid.
terraform_validate ./modules/app_engine 
Success!
 The configuration is valid.
terraform_validate ./modules/budget 
Success!
 The configuration is valid.
terraform_validate ./modules/core_project_factory 
Success!
 The configuration is valid.
terraform_validate ./modules/fabric-project 
Success!
 The configuration is valid.
terraform_validate ./modules/gsuite_enabled 
Provider "gsuite" not available for installation.
A provider named "gsuite" could not be found in the Terraform Registry.
This may result from mistyping the provider name, or the given provider may
be a third-party provider that cannot be installed automatically.
In the latter case, the plugin must be installed manually by locating and
downloading a suitable distribution package and placing the plugin's executable
file in the following directory:
    terraform.d/plugins/linux_amd64
Terraform detects necessary plugins by inspecting the configuration and state.
To view the provider versions requested by each module, run
"terraform providers".
Error: 
no provider exists with the given name
terraform_validate ./modules/gsuite_group 
Success!
 The configuration is valid.
terraform_validate ./modules/project_services 
Success!
 The configuration is valid.
terraform_validate ./modules/quota_manager 
Success!
 The configuration is valid.
terraform_validate ./modules/shared_vpc 
Success!
 The configuration is valid.
terraform_validate ./modules/shared_vpc_access 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/app_engine 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/budget 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/dynamic_shared_vpc 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/fabric_project 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/full 
Provider "gsuite" not available for installation.
A provider named "gsuite" could not be found in the Terraform Registry.
This may result from mistyping the provider name, or the given provider may
be a third-party provider that cannot be installed automatically.
In the latter case, the plugin must be installed manually by locating and
downloading a suitable distribution package and placing the plugin's executable
file in the following directory:
    terraform.d/plugins/linux_amd64
Terraform detects necessary plugins by inspecting the configuration and state.
To view the provider versions requested by each module, run
"terraform providers".
Error: 
no provider exists with the given name
terraform_validate ./test/fixtures/minimal 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/vpc_sc_project 
Success!
 The configuration is valid.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

**Comment by morgante** [comment]
@ericyz Please address linting errors.

File: examples/shared_vpc/main.tf [hunk 1]
```diff
@@ -167,3 +167,12 @@ resource "google_compute_address" "example_address" {
...
   address_type = "INTERNAL"
 }
+
+/******************************************
+  Consumer Quota
+ *****************************************/
+module "project_quota_manager" {
+  source          = "../../modules/quota_manager"
+  project_id      = module.host-project.project_id
+  consumer_quotas = var.consumer_quotas
```
Comment (morgante) [inline examples/shared_vpc/main.tf:177]: We're trying to have examples use hard-coded values where possible, let's put a representative example in here.

File: modules/gsuite_enabled/README.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ The roles granted are specifically:
 | budget\_alert\_spent\_percents | A list of percentages of the budget to alert on when threshold is exceeded | `list(number)` | <pre>[<br>  0.5,<br>  0.7,<br>  1<br>]</pre> | no |
 | budget\_amount | The amount to use for a budget alert | `number` | `null` | no |
 | budget\_monitoring\_notification\_channels | A list of monitoring notification channels in the form `[projects/{project_id}/notificationChannels/{channel_id}]`. A maximum of 5 channels are allowed. | `list(string)` | `[]` | no |
+| consumer\_quotas | The quotas configuration you want to override to the project. | <pre>list(object({<br>    service = string,<br>    metric  = string,<br>    limit   = string,<br>    value   = string,<br>  }))</pre> | `[]` | no |
```
Comment (morgante) [inline modules/gsuite_enabled/README.md:73]:
```suggestion
| consumer\_quotas | The quotas configuration you want to override for the project. | <pre>list(object({<br>    service = string,<br>    metric  = string,<br>    limit   = string,<br>    value   = string,<br>  }))</pre> | `[]` | no |
```

File: modules/quota_manager/main.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  consumer_quotas = { for index, quota in var.consumer_quotas : index => quota }
```
Comment (morgante) [inline modules/quota_manager/main.tf:18]: This should probably be indexed by service+metric. Ordered indexing is brittle.

**Review by morgante — CHANGES_REQUESTED** [review]

**Review by morgante — APPROVED** [review]

**Pull request merged**